packages/core/src/components/forms/_file-upload.scss

```diff
@@ -65,33 +65,24 @@ $file-upload-button-width-large: $pt-grid-size * 8.5;
 
 .pt-file-upload-input {
   @include pt-input();
+  @include overflow-ellipsis();
   position: absolute;
   top: 0;
   right: 0;
   left: 0;
-  margin-right: $file-upload-button-width;
-  border-top-right-radius: 0;
-  border-bottom-right-radius: 0;
-  height: $pt-input-height;
-  line-height: $pt-input-height;
-  color: $pt-text-color-muted;
+  padding-right: $file-upload-button-width + $input-padding-horizontal;
+  color: $pt-placeholder-color;
```

Comment:
unfortunately we have no idea about chosen-file state (see the callout about static file name). but agreed on default color.